master/reference/calicoctl/resources/index.md

```diff
@@ -93,3 +93,25 @@ A file may contain multiple resource documents specified in a YAML list format.
     expectedIPs:
     - 1.2.3.5
 ```
+
+### Namespaced resources
```

Comment:
I would have thought we don't need this section, or the non-namespaced one below.

I think we need:
- A blurb in "NetworkPolicy" which says "The selectors on this policy apply only to workload endpoints which share a namespace"
- A blurb in "GlobalNetworkPolicy" which says "The selectors here apply to all workload endpoints in the cluster"
- Update the calicoctl command reference to mention `--all-namespaces` and `--namespace = foo`

Reply:
Adding the section here was what I understood as the recommendation from @emanic and then link to it from the other sections.

Reply:
If we want a generic section that talks about namespaces as a concept in Calico, then I could buy that, but right now this isn't generic content - it's the details of how WEPs and NPs interact. 

Having the WEP and NP docs link somewhere else in order to understand their behavior feels super wrong to me. Those resources should document their own behavior in their respective docs.

Reply:
I like the idea of defining what a namespace is, exactly, in the overview, which would allow us to use the word "namespace" later on and make it a hyperlink so that people can click to discover what it means. The best single place for this information is probably the overview page. Right now, I have to agree, I don't see anywhere that tells me what is a namespace in the context of Calico in the current content. That is something I think we need to know because this term is quite widely used but it has its own definition in the context of Calico.

Beyond that, I think @caseydavenport is probably right that the specifics of the behavior of namespaces wrt to WEP and NP probably belong in those pages and not the overview.

Reply:
It is possible that @gunjan5 is already working on the calicoctl command updates, so I might check with him first before doing anything there though.

Reply:
I'm not sure what the conclusion was of this thread, but I'm also in favor of a small section explaining what a Calico namespace means

Reply:
In a DM conversation with @emanic and @caseydavenport it was decided to remove the section from here and add what I had here to NetworkPolicy and WorkloadEndpoint.

Reply:
We saw that you were defining namespace to some extent within the tables in the pages of the relevant resources. So, we thought that might be sufficient for now @gunjan5 . However, if you have the time, you could maybe put up a new PR that adds something to the overview, I think that might be a nice addition @gunjan5 .